src/replication.c

```diff
@@ -442,7 +442,7 @@ void replicationFeedSlaves(list *slaves, int dictid, robj **argv, int argc) {
     prepareReplicasToWrite();
 
     /* Send SELECT command to every slave if needed. */
-    if (server.slaveseldb != dictid) {
+    if (dictid != -1 && server.slaveseldb != dictid) {
```

Comment:
please make sure that -1 is passed explicitly (never pass `server.slaveseldb` as `dictid`)
update the comment above the assertion